huaweicloud/services/rds/resource_huaweicloud_rds_instance.go

```diff
@@ -1416,6 +1434,63 @@ func updateRdsInstanceSSLConfig(ctx context.Context, d *schema.ResourceData, cli
 	return configRdsInstanceSSL(ctx, d, client, instanceID)
 }
 
+func updateConfiguration(ctx context.Context, d *schema.ResourceData, client, clientV31 *golangsdk.ServiceClient,
+	instanceID string) (context.Context, error) {
+	if !d.HasChange("param_group_id") {
+		return ctx, nil
+	}
+	if _, ok := d.GetOk("param_group_id"); !ok {
+		return ctx, nil
+	}
+
+	opts := instances.ApplyConfigurationOpts{
+		InstanceIds: []string{instanceID},
+	}
+	log.Printf("[DEBUG] Update opts of RDS configuration: %+v", opts)
+
+	retryFunc := func() (interface{}, bool, error) {
+		res, err := instances.ApplyConfiguration(clientV31, d.Get("param_group_id").(string), opts).Extract()
+		retry, err := handleMultiOperationsError(err)
+		return res, retry, err
+	}
+	res, err := common.RetryContextWithWaitForState(&common.RetryContextWithWaitForStateParam{
+		Ctx:          ctx,
+		RetryFunc:    retryFunc,
+		WaitFunc:     rdsInstanceStateRefreshFunc(client, instanceID),
+		WaitTarget:   []string{"ACTIVE"},
+		Timeout:      d.Timeout(schema.TimeoutUpdate),
+		DelayTimeout: 10 * time.Second,
+		PollInterval: 10 * time.Second,
+	})
+	if err != nil {
+		return ctx, fmt.Errorf("error updating instance configuration: %s ", err)
+	}
+	resp := res.(*instances.ApplyConfigurationResp)
+	if !resp.Success {
+		return ctx, fmt.Errorf("updating instance configuration is unsuccessful")
+	}
+
+	// wait 30 seconds for the instance to enter the modified status, or the modification has been completed
+	// lintignore:R018
+	time.Sleep(30 * time.Second)
+
+	// if the parameters have not changed, it should be set again
+	// if the parameters have changed, it will be modified in the func updateRdsParameters, it is no need to modify here
+	if parameters, ok := d.GetOk("parameters"); ok && !d.HasChange("parameters") {
+		log.Printf("parameters has been changed")
```

Comment:
once the param_group_id is changed, parameters must be all updated. So remove this `if`

Reply:
done